### PR TITLE
add validations and tests

### PR DESCRIPTION
**Pull request by sjdev2212**
In this feature I : 
Added validation to models.
Adde unit specs for all model methods and validations.
Added flash messages in the create actions in all the controllers.
Solved N+1 problem.
Checked linters

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class Post < ApplicationRecord
+  validates :title, presence: true, length: { maximum: 200 }
```
Comment (MahmoudBakr23) [inline app/models/post.rb]: - [x] Please change the validation's maximum number to `250` as the project requirements dictate. 👍 😉

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -11,10 +11,16 @@ def create
...
       comment.save
-      redirect_to user_post_path(@user_id, @post_id)
+      flash[:notice] = 'Comment created'
+
     else
-      redirect_to new_user_post_comment_path(@current_user.id)
+      flash[:notice] = 'Could not create comment'
+
+      # redirect_to user_post_path(@user_id, @post_id)
+      # redirect_to new_user_post_comment_path(@current_user.id)
```
Comment (MahmoudBakr23) [inline app/controllers/comments_controller.rb]: - [x] Please remove unnecessary lines for cleaner code. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @sj1978 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linters are correct
- Professional PR headline, summary, and commit messages
- Well-written unit tests
- n+1 problems solved

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [x] Please fix the MIT link for a completely professional README file

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (rloterh) [reply app/controllers/comments_controller.rb]: This issue is well resolved! ✔️

File: app/models/post.rb (hunk 1 above)
Comment (rloterh) [reply app/models/post.rb]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Anibal, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂


<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**